src/components/structures/auth/SetupEncryptionBody.tsx

```diff
@@ -114,12 +115,18 @@ export default class SetupEncryptionBody extends React.Component<IProps, IState>
         ev.preventDefault();
         const store = SetupEncryptionStore.sharedInstance();
         store.reset();
-    };
-
-    private onResetConfirmClick = (): void => {
-        this.props.onFinished();
-        const store = SetupEncryptionStore.sharedInstance();
-        store.resetConfirm();
+        Modal.createDialog(ResetIdentityDialog, {
+            onReset: () => {
+                // The user completed the reset process - close this dialog
+                this.props.onFinished();
+                this.onDoneClick();
+            },
+            onFinished: () => {
+                // The user cancelled the reset dialog or click away - go back a step
+                this.onResetBackClick();
```

Comment:
onFinished is called in either case isn't it? Is there a risk this will confuse things by clicking back after the dialog is done (also I don't love that we are calling click handlers for reasons other than when a click happened).

Reply:
I've added an `onCancelled` callback, and used that.  I'm not sure if it's necessary, but it's clearer

Reply:
Also changed the code so that it doesn't call the click handlers.

Reply:
Thanks!